### PR TITLE
Distros.tag

### PR DESCRIPTION
**Pull request by fmoral2**
#### Testing ####

Checklist:
1. If your PR changes anything on or related to Jenkins, run it pointing to your branch to make sure it's okay.

2. Verify code lint; we should not have errors.

3. Update the documentation if needed.

4. Update makefile and docker run if adding new tests.

5. Run your tests at least 4 times with all configurations needed and possible.

6. If needed test with different os types.

### CODE REVIEW

**Review by ShylajaDevadiga — DISMISSED** [review]

File: modules/airgap/instance/instance_server.tf [hunk 1]
```diff
@@ -24,7 +24,7 @@ resource "aws_instance" "master" {
   vpc_security_group_ids = [var.sg_id]
   key_name               = var.key_name
   tags = {
-    Name                 = "${var.resource_name}-server${count.index + 1}"
+    Name                 = "${var.resource_name}-distros-server${count.index + 1}"
```
Comment (mdrahman-suse) [inline modules/airgap/instance/instance_server.tf:27]: Suggest adding a new tag like `Team = distros` which will not make the name longer

File: modules/airgap/instance/instance_server.tf [hunk 2]
```diff
@@ -46,7 +46,7 @@ resource "aws_instance" "worker" {
   vpc_security_group_ids = [var.sg_id]
   key_name               = var.key_name
   tags = {
-    Name                 = "${var.resource_name}-worker${count.index + 1}"
+    Name                 = "${var.resource_name}-distros-worker${count.index + 1}"
```
Comment (mdrahman-suse) [inline modules/airgap/instance/instance_server.tf:49]: https://github.com/rancher/distros-test-framework/pull/178/files#r1823336102

File: modules/airgap/instance/instance_server.tf [hunk 3]
```diff
@@ -72,7 +72,7 @@ resource "aws_instance" "bastion" {
   vpc_security_group_ids = [var.sg_id]
   key_name               = var.key_name
   tags = {
-    Name                 = "${var.resource_name}-bastion"
+    Name                 = "${var.resource_name}-distros-bastion"
```
Comment (mdrahman-suse) [inline modules/airgap/instance/instance_server.tf:75]: https://github.com/rancher/distros-test-framework/pull/178/files#r1823336102

File: modules/bastion/instance_server.tf [hunk 4]
```diff
@@ -20,7 +20,7 @@ resource "aws_instance" "bastion" {
   vpc_security_group_ids = [var.sg_id]
   key_name               = var.key_name
   tags = {
-    Name                 = "${var.resource_name}-bastion-server"
+    Name                 = "${var.resource_name}-distros-bastion-server"
```
Comment (mdrahman-suse) [inline modules/bastion/instance_server.tf:23]: https://github.com/rancher/distros-test-framework/pull/178/files#r1823336102

File: modules/rke2/windows_worker/instances_worker.tf [hunk 5]
```diff
@@ -13,7 +13,7 @@ resource "aws_instance" "windows_worker" {
   vpc_security_group_ids = ["${var.sg_id}"]
   key_name               = var.key_name
   tags = {
-    Name = "${var.resource_name}-windows-worker"
+    Name = "${var.resource_name}-distros-windows-worker"
```
Comment (mdrahman-suse) [inline modules/rke2/windows_worker/instances_worker.tf:16]: https://github.com/rancher/distros-test-framework/pull/178/files#r1823336102

File: pkg/testcase/upgradenodereplacement.go [hunk 6]
```diff
@@ -33,7 +33,7 @@ func TestUpgradeReplaceNode(cluster *shared.Cluster, flags *customflag.FlagConfi
 	// create server names.
 	var serverNames, instanceServerIds, newExternalServerIps, newPrivateServerIps []string
 	for i := 0; i < len(cluster.ServerIPs); i++ {
-		serverNames = append(serverNames, fmt.Sprintf("%s-server-replace-%d", resourceName, i+1))
+		serverNames = append(serverNames, fmt.Sprintf("%s-distros-server-replace-%d", resourceName, i+1))
```
Comment (mdrahman-suse) [inline pkg/testcase/upgradenodereplacement.go:36]: https://github.com/rancher/distros-test-framework/pull/178/files#r1823336102 May be add a tag when creating new instance?

File: modules/k3s/master/instances_server.tf [hunk 7]
```diff
@@ -6,7 +6,7 @@ resource "aws_db_instance" "db" {
   engine                 = var.external_db
   engine_version         = var.external_db_version
   instance_class         = var.instance_class
-  db_name                = "mydb"
+  db_name                = "distros-mydb"
```
Comment (mdrahman-suse) [inline modules/k3s/master/instances_server.tf:9]: nit: distros-qa-db?

File: modules/k3s/master/instances_server.tf [hunk 8]
```diff
@@ -46,7 +46,7 @@ resource "aws_eip" "master_with_eip" {
   count                   = var.create_eip ? 1 : 0
   domain                  = "vpc"
   tags                    = {
-    Name ="${var.resource_name}-server1"
+    Name ="${var.resource_name}-distros-server1"
```
Comment (mdrahman-suse) [inline modules/k3s/master/instances_server.tf:49]: https://github.com/rancher/distros-test-framework/pull/178/files#r1823336102

**Review by mdrahman-suse — DISMISSED** [review]
LGTM

**Comment by mdrahman-suse** [comment]
needs rebasing

File: modules/airgap/instance/instance_server.tf [hunk 9]
```diff
@@ -131,3 +131,7 @@ resource "null_resource" "prepare_bastion" {
     ]
   }
 }
+
+locals {
+  resource_tag =  "distros-qa"
+}
```
Comment (mdrahman-suse) [inline modules/airgap/instance/instance_server.tf:137]: nit: new line

File: modules/bastion/instance_server.tf [hunk 10]
```diff
@@ -31,4 +31,8 @@ resource "aws_instance" "bastion" {
   provisioner "local-exec" {
     command = "echo ${aws_instance.bastion[0].public_ip} > /tmp/${var.resource_name}_bastion_ip"
   }
+}
+
+locals {
+  resource_tag =  "distros-qa"
 }
```
Comment (mdrahman-suse) [inline modules/bastion/instance_server.tf:38]: nit: new line

File: modules/k3s/master/instances_server.tf [hunk 11]
```diff
@@ -505,5 +478,17 @@ depends_on = [aws_instance.master, aws_instance.master2-ha]
...
+  total_server_count      = var.no_of_server_nodes + var.etcd_only_nodes + var.etcd_cp_nodes + var.etcd_worker_nodes + var.cp_only_nodes + var.cp_worker_nodes
+  master_node_ip          = var.create_eip ? aws_eip.master_with_eip[0].public_ip : aws_instance.master.public_ip
+  secondary_masters       = var.no_of_server_nodes + var.etcd_only_nodes + var.etcd_cp_nodes + var.etcd_worker_nodes + var.cp_only_nodes + var.cp_worker_nodes - 1
+
+  fqdn                    = var.create_lb ? aws_route53_record.aws_route53[0].fqdn : var.create_eip ? aws_eip.master_with_eip[0].public_ip : "fake.fqdn.value"
+  node_token              = trimspace(data.local_file.token.content)
+
+  random_string           =  random_string.suffix.result
+  resource_tag            =  "distros-qa"
+ }
```
Comment (mdrahman-suse) [inline modules/k3s/master/instances_server.tf:494]: nit: new line

File: modules/rke2/windows_worker/instances_worker.tf (hunk 5 above)
Comment (mdrahman-suse) [inline modules/rke2/windows_worker/instances_worker.tf:16]: `resource_tag` here as well with `distros-qa` to be consistent?

Comment (fmoral2) [reply modules/rke2/windows_worker/instances_worker.tf:16]: sure, forgot

File: modules/rke2/worker/instances_worker.tf [hunk 12]
```diff
@@ -99,3 +97,7 @@ resource "null_resource" "worker_eip" {
   depends_on = [aws_eip.worker_with_eip,
                  aws_eip_association.worker_eip_association]
 }
+
+locals {
+  resource_tag    =  "distros-qa"
+}
```
Comment (mdrahman-suse) [inline modules/rke2/worker/instances_worker.tf:103]: nit: new line

**Review by mdrahman-suse — APPROVED** [review]
One nit, else LGTM

**Review by ShylajaDevadiga — APPROVED** [review]
LGTM

**Pull request merged**